src/openforms/submissions/tests/form_logic/test_modify_components.py

```diff
@@ -172,6 +172,88 @@ def test_hiding_component_empties_its_data(self):
         self.assertEqual(configuration, expected)
         self.assertEqual("", submission_step.data["component2"])
 
+    def test_hiding_component_with_default_value_empties_its_data(self):
```

Comment:
This is not a good regression test. We need a regression test that replicates what happens in the "how to reproduce" part of the ticket.

Additionally, regression tests should always have a reference back to the ticket, the preferred way is to use the test tag: `@tag("gh-1755")` so we can easily run and isolate the test and see why this test exists.